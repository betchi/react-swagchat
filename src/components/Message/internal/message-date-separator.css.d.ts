export const root: string;
export const separate: string;
export const date: string;
export const clear: string;
