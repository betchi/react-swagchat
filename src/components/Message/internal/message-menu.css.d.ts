export const colors: string;
export const colorPrimary: string;
export const colorDark: string;
export const root: string;
export const bottom: string;
export const top: string;
export const imageMenuRoot: string;
export const on: string;
export const off: string;
