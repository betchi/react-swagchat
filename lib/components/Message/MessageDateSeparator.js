"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
exports.MessageDateSeparator = function (props) { return (React.createElement("div", { className: "message-date-separator-root" },
    React.createElement("div", null,
        React.createElement("div", { className: "message-date-separator-separate" },
            React.createElement("div", { className: "message-date-separator-date" }, props.date)),
        React.createElement("div", { className: "message-date-separator-clear" })))); };
//# sourceMappingURL=MessageDateSeparator.js.map