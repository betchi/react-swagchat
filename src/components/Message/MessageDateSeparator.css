.sc-message-date-separator-root {
  width: 100%;
  padding-top: 20px;
}
.sc-message-date-separator-separate {
  width: 100%;
  border-top: 1px solid #eee;
  display: flex;
  flex-direction: column;
  align-items: center;
}
.sc-message-date-separator-date {
  font-size: 0.6em;
  color: #212121;
  width: auto;
  word-break: break-all;
  background-color: #fff;
  border: 1px solid #eee;
  border-radius: 15px;
  padding: 5px 10px;
  margin-top: -12px;
  font-weight: 100;
}
.sc-message-date-separator-clear {
  clear: both;
  margin-bottom: 30px;
};