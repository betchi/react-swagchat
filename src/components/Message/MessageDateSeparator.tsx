import * as React from 'react';

export interface IMessageDateSeparatorProps {
  date: string;
}

export const MessageDateSeparator = (props: IMessageDateSeparatorProps) => (
  <div className="message-date-separator-root">
    <div>
      <div className="message-date-separator-separate">
        <div className="message-date-separator-date">{props.date}</div>
      </div>
      <div className="message-date-separator-clear" />
    </div>
  </div>
);