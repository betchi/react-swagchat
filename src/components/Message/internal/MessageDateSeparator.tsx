import * as React from 'react';
import * as styles from './message-date-separator.css';
const classNames = require('classnames');

export interface IMessageDateSeparatorProps {
  date: string;
  className?: string;
  style?: Object;
}

export class MessageDateSeparator extends React.Component<IMessageDateSeparatorProps, {}> {
  public static defaultProps: Partial<IMessageDateSeparatorProps> = {
    className: '',
    style: {},
  };

  render(): JSX.Element  {
    const { date, className, style} = this.props;
    return (
      <div
        className={classNames(styles.root, className)}
        style={style}
      >
        <div>
          <div className={styles.separate}>
            <div className={styles.date}>{date}</div>
          </div>
          <div className={styles.clear} />
        </div>
      </div>
    );
  }
}