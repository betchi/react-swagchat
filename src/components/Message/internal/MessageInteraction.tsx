import * as React from 'react';
import {
  IPluginMessage,
  ISettingState,
  IUser,
  IRoom,
} from 'swagchat-sdk';

export interface IMessageInteractionProps {
  settingState: ISettingState;
  user: IUser;
  room: IRoom;
  position: 'top' | 'bottom';
  isAlwaysDisplay: boolean;
  onTextareaFocus: () => void;
  onTextareaBlur: () => void;
  pluginMessages: IPluginMessage[];
  customPluginMessages: IPluginMessage[];
  currentMenuIndex: number;
  availableMessageTypes: string[] | null;
}

export class MessageInteraction extends React.Component<IMessageInteractionProps, {}> {
  public static defaultProps: Partial<IMessageInteractionProps> = {
    position: 'bottom',
  };

  renderMessageInteraction = () => {
    const { settingState, pluginMessages, isAlwaysDisplay, currentMenuIndex, user, room, position, onTextareaBlur, onTextareaFocus } = this.props;
    let interactions = new Array;
    pluginMessages.map((pluginMessage, i) => {
      if ((pluginMessage.position === position && pluginMessage.isAlwaysDisplay) ||
        (pluginMessage.position === position && i === currentMenuIndex)) {
          interactions.push(React.createElement(
            pluginMessage.interaction, {
              key: 'plugin-custom-message-interaction-' + i,
              settingState: settingState,
              user: user,
              room: room,
              position: position,
              isAlwaysDisplay: isAlwaysDisplay,
              onTextareaFocus: onTextareaFocus,
              onTextareaBlur: onTextareaBlur,
            }
          ));
        }
    });

    return interactions;
  }

  renderCustomMessageInteraction = () => {
    const { settingState, customPluginMessages, isAlwaysDisplay, currentMenuIndex, user, room, position, onTextareaBlur, onTextareaFocus } = this.props;
    let interactions = new Array;
    customPluginMessages.map((pluginMessage, i) => {
      if ((pluginMessage.position === position && pluginMessage.isAlwaysDisplay) ||
        (pluginMessage.position === position && i === currentMenuIndex)) {
        interactions.push(React.createElement(
          pluginMessage.interaction, {
            key: 'plugin-custom-message-interaction-' + i,
            settingState: settingState,
            user: user,
            room: room,
            position: position,
            isAlwaysDisplay: isAlwaysDisplay,
            onTextareaFocus: onTextareaFocus,
            onTextareaBlur: onTextareaBlur,
          }
        ));
      }
    });
    return interactions;
  }

  render(): JSX.Element {
    const {
      position,
      availableMessageTypes,
    } = this.props;

    let interactions;
    if (availableMessageTypes && availableMessageTypes.length > 0) {
      interactions = this.renderCustomMessageInteraction();
    } else {
      interactions = this.renderMessageInteraction();
    }

    const classNames = require('classnames');
    return (
      <div className={interactions.length > 0 ? classNames('message-interaction-root', position) : ''}>
        {interactions}
      </div>
    );
  }
}
