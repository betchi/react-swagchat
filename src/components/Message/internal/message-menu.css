@value colors: "../../../styles/theme/default/color.css";
@value color-primary, color-disable from colors;

.root.bottom {
  background-color: #fff;
  overflow-x: auto;
  overflow-y: hidden;
  white-space: nowrap;
  -webkit-overflow-scrolling: touch;
  display: flex;
  justify-content: flex-start;
  align-items: center;
}
.root.top {
  overflow-x: auto;
  overflow-y: hidden;
  white-space: nowrap;
  -webkit-overflow-scrolling: touch;
  display: flex;
  justify-content: space-around;
  align-items: center;
  height: 3rem;
  background-color: rgba(200,200,200,0.5);
}
.root.top .image-menu-root {
  height: 2rem;
}
.root.top i {
  font-size: 2rem;
}
.on {
  color: color-primary;
}
.off {
  color: color-disable;
}
.off:hover {
  color: color-primary;
}