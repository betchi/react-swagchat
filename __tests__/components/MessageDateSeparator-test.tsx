declare var jest, describe, it, expect, require;

import * as React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import { MessageDateSeparator } from '../../src/components/Message/internal/MessageDateSeparator';

const testComponent1 = shallow(
  <MessageDateSeparator
    date="1/1"
  />
);
it('MessageDateSeparator renders correctly', () => {
  const j = toJson(testComponent1);
  expect(j).toMatchSnapshot();
});

const testComponent2 = shallow(
  <MessageDateSeparator
    date="1/1"
    className="test-class"
    style={{width: '100px'}}
  />
);
it('MessageDateSeparator renders correctly', () => {
  const j = toJson(testComponent2);
  expect(j).toMatchSnapshot();
});
