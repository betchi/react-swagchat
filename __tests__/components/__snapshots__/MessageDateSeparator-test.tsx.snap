// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageDateSeparator renders correctly 1`] = `
<div
  className=""
  style={Object {}}
>
  <div>
    <div>
      <div>
        1/1
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`MessageDateSeparator renders correctly 2`] = `
<div
  className="test-class"
  style={
    Object {
      "width": "100px",
    }
  }
>
  <div>
    <div>
      <div>
        1/1
      </div>
    </div>
    <div />
  </div>
</div>
`;
